### DISCUSSION TITLE
Model for a nested dictionary in a list

### DISCUSSION THREAD

**Discussion by mohammadali95** [pydantic/pydantic#8081] [Question]
I am trying to see if I can have a Pydantic model for the following data input:
`request = {'Phones': {'Phone': [{'Type': 'home', 'MobileCarrier': '', 'AllowText': 'False', 'AllowAutoDial': 'False'}, 
{'Type': 'home', 'MobileCarrier': '', 'AllowText': 'False', 'AllowAutoDial': 'False'}]}}`

My main model looks like the following:
```
class InputDataModel(BaseModel):
    Phones: Phone = Field(default_factory=dict)
```

And I had defined my phone model as: 
```
class Phone(BaseModel):
    Type: Optional[Literal['work', 'cell', 'home']] = ''
    AllowText: Optional[Literal["False", "True"]] = 'False'
    AllowAutoDial: Optional[Literal["False", "True"]] = 'False'
    MobileCarrier: Optional[str] = ''
```

But I would like the output structure similar to the input after validation:
` {'Phones': {'Phone': [{'Type': 'home', 'MobileCarrier': '', 'AllowText': 'False', 'AllowAutoDial': 'False'}]}}`

Any thoughts on how to achieve this?

**Comment by jamesjnadeau** [comment]
Did you ever solve this? facing the same problem, thanks!

**Reply by mohammadali95** [reply]
Hey @jamesjnadeau, 

Yes - you just have to have another nested model within your main model which represents the list of dicts which in this case the list of Phone. Below is how you can achieve this:

```
from pydantic import BaseModel, Field
from typing import Optional, Literal


class Phone(BaseModel):
    Type: Optional[Literal['work', 'cell', 'home']] = ''
    AllowText: Optional[Literal["False", "True"]] = 'False'
    AllowAutoDial: Optional[Literal["False", "True"]] = 'False'
    MobileCarrier: Optional[str] = ''

class PhoneList(BaseModel):
    Phone: list[Phone]


class InputDataModel(BaseModel):
    Phones: PhoneList
```

I hope this helps - let me know if I can help in any other way!
